aten/src/ATen/core/Device.cpp

```diff
@@ -18,7 +18,32 @@ std::pair<Device::Type, size_t> parse_type(const std::string& device_string) {
   if (position != std::string::npos) {
     return {Device::Type::CUDA, 4};
   }
-  AT_ERROR("Expected 'cpu' or 'cuda' device type at start of device string");
+  position = device_string.find("mkldnn");
+  if (position != std::string::npos) {
+    return {Device::Type::MKLDNN, 6};
+  }
+  position = device_string.find("mkldnn");
```

Comment:
I think you didn't want to test this twice.

Reply:
fixed